interfaces/attrs.go

```diff
@@ -0,0 +1,197 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package interfaces
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/snap"
+)
+
+type PlugData struct {
+	plug         *snap.PlugInfo
+	dynamicAttrs map[string]interface{}
+}
+
+type SlotData struct {
+	slot         *snap.SlotInfo
```

Comment:
Just think aloud, should this be a `SlotInfo` or is a `SlotRef` sufficient?

Reply:
No, it wouldn't bu sufficient, I need access to slot's and plug's attributes.